addons/mail/controllers/discuss.py

```diff
@@ -443,10 +444,22 @@ def mail_thread_data(self, thread_model, thread_id, request_list, **kwargs):
 
     @http.route('/mail/thread/messages', methods=['POST'], type='json', auth='user')
     def mail_thread_messages(self, thread_model, thread_id, max_id=None, min_id=None, limit=30, **kwargs):
+        # Need a sudo because mail.mail has different ACL than mail.message
+        filtered_ids = request.env['mail.message'].sudo().search(
+            AND([
```

Comment:
you can simplify it a bit like you did with `_mail_thread_messages_filter`, as you prefer

Reply:
As discussed, now the method _mail_thread_messages_filter returns a domain and I use operator on them (so this one cannot be simplified).